domain/schema_validator.go

```diff
@@ -209,7 +220,35 @@ func (s *schemaValidator) Check(txnTS uint64, schemaVer int64, relatedTableIDs [
 
 func (s *schemaValidator) enqueue(schemaVersion int64, relatedTableIDs []int64) {
 	s.deltaSchemaInfos = append(s.deltaSchemaInfos, deltaSchemaInfo{schemaVersion, relatedTableIDs})
-	if len(s.deltaSchemaInfos) > maxNumberOfDiffsToLoad {
+	s.notMergeCnt++
+
+	maxCnt := int(variable.GetMaxDetalSchemaCount())
+	if len(s.deltaSchemaInfos) > maxCnt && s.notMergeCnt > maxCnt/2 {
+		s.merge()
+		s.notMergeCnt = 1
```

Comment:
Why set to 1 here? I think we don't know how many `deltaSchemaInfos` merged here.

Reply:
We call `merge` before setting the value. So I think that the previous `deltaSchemaInfos` have been merged.